service/common/mapper/searchAttribute.go

```diff
@@ -45,7 +45,8 @@ func MapToInternalSearchAttributes(attributes []iwfidl.SearchAttribute) (map[str
 
 func MapCadenceToIwfSearchAttributes(searchAttributes *shared.SearchAttributes, requestedSearchAttributes []iwfidl.SearchAttributeKeyAndType) (map[string]iwfidl.SearchAttribute, error) {
 	if searchAttributes == nil || len(requestedSearchAttributes) == 0 {
-		return nil, nil
+		// return empty map rather than nil
```

Comment:
```
unable to convert value <nil> to type KEYWORD_ARRAY for key IwfExecutingStateIds
```
The error was when we converting a nil to KeywordArray, so I believe it's at line 66...

You should be able to reproduce this, if you modify `wait_until_search_attributes_optimization_test.go` to make it more complex -- for the default mode (ENABLED_FOR_STATES_WITH_WAIT_UNTIL_ONLY) , after the S7-1, start another S8 with waitUntil and it will try to refresh `IwfExecutingStateIds` with null value...

![Screenshot 2024-12-03 at 2 41 02 PM](https://github.com/user-attachments/assets/a97eae39-752c-4b52-92fa-78f9d1b04daf)

Reply:
Basically, whenever it tried to refresh `IwfExecutingStateIds` that is nil. At the beginning this SA doesn't even exist so there wasn't error. 

Then it was set to some values(array), and then cleared up as a nil. Then setting again will see the errors.

Reply:
An easy way to produce this:
S1->S2->S1 where S1 has waitUntil but S2 doesn't have.

Reply:
Fixed + made an existing test to verify it works as expected